shell/components/firebase-upload.js

```diff
@@ -0,0 +1,62 @@
+import Xen from './xen/xen.js';
+import Firebase from './firebase.js';
+
+const {firebase, storage} = Firebase;
+
+const template = `
+  <style>
+    ::slotted(*) {
+      cursor: pointer;
+    }
+    input[type="file"] {
+      display: none;
+    }
+  </style>
+  <label>
+    <slot></slot>
+    <input type="file" accept="{{accept}}" multiple="{{multiple}}" on-change="_onFilesChanged">
+  </label>
+`;
+
+class FirebaseUpload extends Xen.Base {
```

Comment:
Add a comment noting this is currently low fidelity and more logic will shift into runtime over time? Something so that someone new reading it has rough orientation.